yt_dlp/extractor/bbc.py

```diff
@@ -791,6 +791,17 @@ class BBCIE(BBCCoUkIE):  # XXX: Do not subclass from concrete IE
             'timestamp': 1638230731,
             'upload_date': '20211130',
         },
+    }, {
+        # video with script id __NEXT_DATA__ and value as JSON string
+        'url': 'https://www.bbc.com/news/uk-68546268',
```

Comment:
The metadata is not being extracted from the UK page. I'll check what needs to be changed.

Reply:
This is what we get from bbc.co.uk/programmes/p0hj0lq7:
```json
  "id": "p0hj0lq7",
  "title": "BBC - Nasser Hospital doctor describes his treatment by IDF",
  "description": "Nasser Hospital doctor describes his treatment by IDF",
  "thumbnail": "https://ichef.bbci.co.uk/images/ic/1200x675/p0hj0n63.jpg",
  "duration": null,
```
In the UK page, the hydration JSON string in `__INITIAL_DATA__` has `.data` containing a member whose value is an object with `.name` `article` and whose `.data.cǒntent.model.blocks` is a list including the target object with `.type` `video`. This doesn't contain the metadata that the PR finds in the US page.

There is also `.stores.article.articleBodyContent` containing a member with `.type` `video` and this is much more productive:
<details><summary><code>articleBodyContent</code> video object</summary>

```py
        {
          'type': 'video',
          'model': {
            'locator': 'urn:bbc:pips:pid:p0hhzw5j',
            'blocks': [
              {
                'type': 'aresMedia',
                'model': {
                  'blocks': [
                    {
                      'type': 'aresMediaMetadata',
                      'blockId': 'urn:bbc:ares::clip:p0hhzw5j',
                      'model': {
                        'advertising': false,
                        'caption': 'Dr Abu Sabha said he was detained and beaten by Israeli forces after the raid on Nasser Hospital.',
                        'embedding': true,
                        'format': 'video',
                        'id': 'p0hhzw5j',
                        'imageCopyright': 'BBC',
                        'imageUrl': 'ichef.bbci.co.uk/images/ic/$recipe/p0hj0n63.jpg',
                        'smpKind': 'programme',
                        'subType': 'clip',
                        'syndication': {
                          'destinations': [
                          ]
                        },
                        'synopses': {
                          'short': 'Nasser Hospital doctor describes his treatment by IDF',
                          'long': 'Doctor Abu Sabha said he was detained by Israeli forces after the raid on Nasser Hospital and feared for his life.\n\nThe IDF said "during the activity, about 200 terrorists and suspects of terrorist activity were detained, including some who posed as medical teams, many weapons were found, as well as closed medicines intended for Israeli hostages."',
                          'medium': 'Doctor Abu Sabha said he was detained by Israeli forces after the raid on Nasser Hospital.'
                        },
                        'title': 'Nasser Hospital doctor describes his treatment by IDF',
                        'versions': [
                          {
                            'versionId': 'p0hj0lq7',
                            'types': [
                              'Editorial'
                            ],
                            'duration': 104,
                            'durationISO8601': 'PT1M44S',
                            'warnings': {
                              'short': 'Contains some upsetting scenes.',
                              'long': 'Contains some scenes which some viewers may find upsetting.'
                            },
                            'availableTerritories': {
                              'uk': true,
                              'nonUk': true
                            },
                            'availableFrom': 1710188248000
                          }
                        ]
                      }
                    }
                  ]
                }
              },
```

</details>

This object parallels the `video_data` extracted in the PR but the metadata in the `.model` includes `duration` (and `timestamp` directly). If `duration` can't be extracted from the US page, the `__INITIAL_DATA__` code should get the `duration` but comment it out of the test: it will then fail in UK unless the comment is removed.

Reply:
I coded up a UK version. I think the right way forward is to add an inline function after l.1232 (ie, just before the search for `initial_data`) using the code from l.1281 onward, like this:
```py
        def parse_model(model):
            '''Extract single video from model structure'''
            item_id = traverse_obj(model, ('versions', 0, 'versionId', {str}))
            if not item_id:
                return
            formats, subtitles = self._download_media_selector(item_id)
            return {
                'id': item_id,
                'formats': formats,
                'subtitles': subtitles,
                **traverse_obj(model, {
                    'title': ('title', {str}),
                    'thumbnail': ('imageUrl', {lambda u: urljoin(url, u.replace('$recipe', 'raw'))}),
                    'description': (
                        'synopses', ('long', 'medium', 'short'), {str}, any),
                    'duration': ('versions', 0, 'duration', {int}),
                    'timestamp': ('versions', 0, 'availableFrom', {lambda x: int_or_none(x, scale=1000)}),
                })
            }
```
This could then be used to process the `aresMediaMetadata` model from the UK page as well as the `mediaMetadata` model in the PR. The separately extracted `timestamp` would have to be added to the returned data, as posted in a separate suggestion.

Reply:
ok, it seems that the uk use can be done in a separate PR, but I have added this function so it can be used for that use case eventually.

Reply:
Once this is ready to go for US, I can run the tests in the UK, which may throw up some issues, probably mostly dead pages that we can ignore (`'skip': '404'`) for now.

If desired, I can propose a diff to apply the UK changes so that they can be included here, or as you say it could be a separate PR (but that might end up waiting for me to implement it upstream).

Reply:
I can go ahead and take in the changes here.